docs/source/miscellaneous/dropwizard-metrics.rst

```diff
@@ -0,0 +1,61 @@
+Cassandra node in your cluster. For more details on what information each type of metric provides, we recommend reading
+the Metrics `Getting Started Guide <http://metrics.dropwizard.io/3.1.0/getting-started/#>`__.
+
+**Gauges**
+
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.numBlacklistedHosts``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.<host>.requestConnectionExceptionProportion``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.<host>.requestFailureProportion``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.requestConnectionExceptionProportion``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.requestFailureProportion``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.<host>.meanActiveTimeMillis``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.<host>.meanBorrowWaitTimeMillis``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.<host>.meanIdleTimeMillis``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.<host>.proportionDestroyedByBorrower``
+ - ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer.<host>.proportionDestroyedByEvictor``
+
+**Histograms**
+
+- ``com.palantir.atlasdb.transaction.impl.SnapshotTransaction.byteSizeTx``
+
+**Meters**
+
+- ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.<host>.requestConnectionExceptions``
+- ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.<host>.requestExceptions``
+- ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.<host>.requests``
+- ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.requestConnectionExceptions``
+- ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.requestExceptions``
+- ``com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool.requests``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskReadOnly.failures``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskReadOnly.failures.java.lang.IllegalArgumentException``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskWithRetry.failures``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskWithRetry.failures.feign.FeignException``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskWithRetry.failures.java.lang.IllegalArgumentException``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskWithRetry.failures.java.lang.IllegalStateException``
+
+**Timers**
+
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.getLockService``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskReadOnly``
+- ``com.palantir.atlasdb.transaction.api.LockAwareTransactionManager.runTaskWithRetry``
```

Comment:
I actually don't think we're adding this directly in AtlasDB yet, this is likely from specific app instrumenting result of TransactionManagers.create in their own app. The main reason we don't do this in AtlasDB yet is the messiness with the return type being a SerializableTransactionManager which is a concrete type rather than interface of LockAwareTransactionManager. Same issue applies to Transactions themselves since SerializableTransaction and SnapshotTransaction are concrete types now